tests/test_artifacts/auth_testing_contract/src/main.sw

```diff
@@ -0,0 +1,50 @@
+contract;
+
+use std::address::Address;
+use std::chain::auth::*;
+use std::contract_id::ContractId;
+use auth_testing_abi::*;
+use std::result::*;
+use std::assert::assert;
+
+impl AuthTesting for Contract {
+    fn is_caller_external() -> bool {
+        caller_is_external()
+    }
+
+    fn returns_msg_sender(expected_id: ContractId) -> bool {
+        let result: Result<Sender, AuthError> = msg_sender();
+        let mut ret = false;
+        if result.is_err() {
+            ret = false;
+        } else {
+            let unwrapped = result.unwrap();
+            if let Sender::ContractId(v) = unwrapped {
+                assert(v == expected_id);
```

Comment:
This test isn't perfect because of this assert which shouldn't be here, but the compiler panics if removed. Could be a compiler bug, will investigate later.

Reply:
cc @otrho

Reply:
This is a bit contrived... ideally the test contracts would just pass the Result returned by `msg_sender()` along to the rust tests and compare the returned values there.
But, I was having issues with contracts returning nested struct/enums (`MissingData` error when using `abigen~ macro) and so I simplified the return values and make comparisons in the contracts themselves.
It feels hacky but works...